pkg/daemon/daemon.go

```diff
@@ -1470,9 +1470,16 @@ func (dn *Daemon) removeUpdateInProgressTaint(ctx context.Context) error {
 	})
 }
 
-func (dn *Daemon) isLayeredNode() bool {
+func (dn *Daemon) isLayeredNode() *CoreOSDaemon {
 	_, ok := dn.node.Annotations[constants.DesiredImageConfigAnnotationKey]
```

Comment:
I think actually the controller should not allow this situation to be set up.  It also makes sense to have a last-ditch sanity check on the daemon side.

I wouldn't conflate checking "has a desired image config annotation" with "is coreos" check though, they seem like separate things.

Reply:
Would `isLayeredNode -> shouldUpdateLayered` clarify that? Or is that squashing checks together unclearly

Reply:
Yeah,  `isLayeredNode` is more "has this node been assigned a desired image configuration", which in our human description makes it a "layered node" but it's more a quality of the update being requested rather than a quality of the node. 

If we had `canUpdateLayered()` in the daemon and one of the sanity checks in there was "is coreos" I think that would make sense? 


As an aside, should we just call it an "Image update"? Then we have "image update" vs "machineconfig update" and it's just referring to the type of the object in `spec.Configuration` ? (Unless you think we want a new CRD for holding whatever a layered update looks like, kinda like machineconfig... :stuck_out_tongue: ) 

(And yeah, I'm well aware I had a hand in naming all this stuff how it is :smile: )

Reply:
Yeah I think image update makes sense! Machine config vs image is a lot more clear than what I've been saying (legacy vs layered)

Reply:
> As an aside, should we just call it an "Image update"? Then we have "image update" vs "machineconfig update" and 

Yes!  SGTM